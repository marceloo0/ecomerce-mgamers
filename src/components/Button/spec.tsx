import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'
import { FaAdobe } from 'react-icons/fa'

import { Button } from '.'

describe('<Button />', () => {
  it('should render the medium size by default', () => {
    const { container } = renderWithTheme(<Button>Comprar Agora</Button>)

    expect(screen.getByText(/Comprar Agora/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the small size', () => {
    renderWithTheme(<Button size="small">Comprar Agora</Button>)

    expect(screen.getByText(/Comprar Agora/i)).toBeInTheDocument()
  })

  it('should render the large size', () => {
    renderWithTheme(<Button size="large">Comprar Agora</Button>)

    expect(screen.getByText(/Comprar Agora/i)).toBeInTheDocument()
  })

  it('should render an icon version', () => {
    renderWithTheme(
      <Button icon={<FaAdobe data-testid="icon" />}>Comprar Agora</Button>
    )

    expect(screen.getByText(/Comprar Agora/i)).toBeInTheDocument()
    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })
})
