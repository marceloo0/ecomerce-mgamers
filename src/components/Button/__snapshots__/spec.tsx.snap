// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the medium size by default 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 1px solid transparent;
  border-radius: 0.4rem;
  cursor: pointer;
  color: #C7ECEC;
  background-color: #201C2B;
  padding: 1rem;
}

.c0:hover {
  background-color: #332756;
}

<button
  class="c0"
>
  <span>
    Comprar Agora
  </span>
</button>
`;
